source/FreeRTOS_DNS_Parser.c

```diff
@@ -1243,10 +1230,14 @@
                 /* This function will fill in the eth addresses and send the packet */
                 vReturnEthernetFrame( pxNetworkBuffer, pdFALSE );
 
-                if( pxNewBuffer != NULL )
-                {
-                    vReleaseNetworkBufferAndDescriptor( pxNewBuffer );
-                }
+/*pxNewBuffer and pxNetworkBuffer are now the same pointers.pxNewBuffer is always NULL after re-allocation*/
+
+/*
+ *              if( pxNewBuffer != NULL )
+ *              {
+ *                vReleaseNetworkBufferAndDescriptor( pxNewBuffer );
+ *              }
+ */
```

Comment:
Can you elaborate more about why pxNewBuffer is always NULL?
It's used in different APIs as network buffer and it's weird to be NULL.

Reply:
Updated the comment, pxNewBuffer and pxNetworkBuffer are the same buffer and not duplicated, hence it is enough to Release the buffer once, no need for the If condition check